examples/spec/integration/register_namespace_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'workflows/timeout_workflow'
+
+describe 'Temporal.register_namespace' do
+  it 'can register a new namespace' do
+    # have to generate a new namespace on each run because currently can't delete namespaces
+    name = "test_namespace_#{SecureRandom.uuid}"
+    description = 'this is the description'
+    retention_period = 30
+    namespace_data = { test: 'value' }
+
+    result = Temporal.register_namespace(name, description, retention_period: retention_period, namespace_data: namespace_data)
+    expect(result).to be_an_instance_of(Temporal::Api::WorkflowService::V1::RegisterNamespaceResponse)
+
+    # fetch the namespace from Temporal and check it exists and has the correct settings
+    found_namespace = nil
+    next_page_token = ''
+
+    while found_namespace.nil?
+      result = Temporal.list_namespaces(page_size: 100, next_page_token: next_page_token)
+      result.namespaces.each do |namespace|
+        if namespace.namespace_info.name == name
+          found_namespace = namespace
+          break
+        end
```

Comment:
if `namespaces` is enumerable, then just use `find` directly, e.g.
```suggestion
      found_namespace = result.namespaces.find do |namespace|
        namespace.namespace_info.name == name
        end
```

Reply:
nifty, sorry ruby isn't my main language! love this tho and will commit it